rust/src/ffi.rs

```diff
@@ -1,34 +1,24 @@
 use bdk::bitcoin::secp256k1::Secp256k1;
-use std::collections::HashSet;
-//use bdk::bitcoin::secp256k1::rand as bdk_rand;
-use crate::types::{DatabaseConfig, ExtendedKeyInfo};
+use crate::types::{DatabaseConfig, BlockTime, TransactionDetails, AddressIndex, AddressInfo, OutPoint, TxOut, LocalUtxo, Balance};
 use crate::utils::config_database;
-use bdk::bitcoin::hashes::hex::ToHex;
+use bdk::bitcoin::hashes::hex::{FromHex, ToHex};
 use bdk::bitcoin::util::bip32::DerivationPath as BdkDerivationPath;
 use bdk::bitcoin::util::psbt::PartiallySignedTransaction;
-use bdk::bitcoin::{Address, Network, OutPoint as BdkOutPoint, Script, Txid};
+use bdk::bitcoin::{Address as BdkAddress,Script as BdkScript, Network, OutPoint as BdkOutPoint, Txid};
+
 use bdk::blockchain::any::AnyBlockchain;
 use bdk::database::{AnyDatabase, AnyDatabaseConfig, ConfigurableDatabase};
-use bdk::descriptor::DescriptorXKey;
+use bdk::descriptor::{DescriptorXKey, ExtendedDescriptor};
 use bdk::keys::bip39::{Language, Mnemonic, WordCount};
-use bdk::keys::{
-    DerivableKey, DescriptorPublicKey as BdkDescriptorPublicKey,
-    DescriptorSecretKey as BdkDescriptorSecretKey, ExtendedKey, GeneratableKey, GeneratedKey,
-};
+use bdk::keys::{DerivableKey, DescriptorPublicKey as BdkDescriptorPublicKey, DescriptorSecretKey as BdkDescriptorSecretKey, ExtendedKey, GeneratableKey, GeneratedKey};
```

Comment:
Yes, possibly this may not be required. Worth trying it out. In the past I have used BDK directly.
In the current version I wanted to reuse most of bdk-ffi/lib.rs as is